pkg/pss/pss_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pss_test
+
+import (
+	"context"
+	"io/ioutil"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/localstore"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/pss"
+	"github.com/ethersphere/bee/pkg/pusher"
+	"github.com/ethersphere/bee/pkg/pushsync"
+	pushsyncmock "github.com/ethersphere/bee/pkg/pushsync/mock"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+	mocktopology "github.com/ethersphere/bee/pkg/topology/mock"
+	"github.com/ethersphere/bee/pkg/trojan"
+)
+
+// Wrap the actual storer to intercept the modeSet that the pusher will call when a valid receipt is received
+type Store struct {
+	storage.Storer
+	modeSet   map[string]storage.ModeSet
+	modeSetMu *sync.Mutex
+}
+
+// TestTrojanChunkRetrieval creates a trojan chunk
+// mocks the localstore
+// calls pss.Send method and verifies it's properly stored
+func TestTrojanChunkRetrieval(t *testing.T) {
+	var err error
```

Comment:
you can make the test more readable by either creating a helper method that will bootstrap the dependencies:
`ctx, pss, target, targets(???), payload := newTestPss()` 
or
```go
var (
   ctx = context.Background()
   testTags = tags.NewTags()
   pushSyncService = pushsyncmock.New(...)
   etc...
)
```